parsers/US_IPC.py

```diff
@@ -0,0 +1,151 @@
+#!usr/bin/env python3
+
+"""Parser for the Idaho Power Comapny area of the United States."""
+
+from dateutil import parser, tz
+from itertools import groupby
+from logging import getLogger
+import requests
+
+
+# NOTE No pumped storage yet but future ideas can be found at the following url.
+# https://docs.idahopower.com/pdfs/AboutUs/PlanningForFuture/irp/IRP.pdf
+
+# Renewable energy (PURPA) is likely bought with credits from outside the Utility
+# area and not supplied to customers. For that reason those types are commented out. 
+
+PRODUCTION_URL = 'https://api.idahopower.com/Energy/Api/V1/GenerationAndDemand/Subset'
+
+GENERATION_MAPPING = {#'Non-Utility Geothermal': 'geothermal',
+                      'Hydro': 'hydro',
+                      'Coal': 'coal',
+                      'Diesel': 'oil',
+                      #'PURPA/Non-Utility Wind': 'wind',
+                      'Natural Gas': 'gas'
+                      #'PURPA/Non-Utility Solar': 'solar',
+                      #'PURPA Other': 'biomass'
+                      }
+
+TIMEZONE = tz.gettz("America/Boise")
+
+
+def get_data(session=None):
+    """Returns a list of dictionaries."""
+
+    s = session or requests.Session()
+
+    req = requests.get(PRODUCTION_URL)
+    json_data = req.json()
+    raw_data = json_data['list']
+
+    return raw_data
+
+
+def timestamp_converter(timestamp):
+    """Converts a str timestamp into an aware datetime object."""
+
+    dt_naive = parser.parse(timestamp)
+    dt_aware = dt_naive.replace(tzinfo=TIMEZONE)
```

Comment:
FYI, we usually use `arrow.get(timestamp, "America/Boise")` as it's a bit simpler, but it doesn't change the result